### PR TITLE
ogre: 1.12.1 -> 1.12.9, cegui: enable build on darwin

### PR DESCRIPTION
**Pull request by siraben**
###### Description of changes
Update and fix darwin build.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/cegui/default.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, cmake, ogre, freetype, boost, expat }:
+{ stdenv, fetchurl, lib, cmake, ogre, freetype, boost, expat, libiconv }:
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/cegui/default.nix:1]:
```suggestion
{ stdenv, fetchurl, cmake, ogre, freetype, boost, expat, libiconv }:
```

File: pkgs/development/libraries/cegui/default.nix [hunk 2]
```diff
@@ -10,12 +10,13 @@ stdenv.mkDerivation rec {
   };
 
 
-  buildInputs = [ cmake ogre freetype boost expat ];
+  buildInputs = [ cmake ogre freetype boost expat ]
+                ++ lib.optionals stdenv.isDarwin [ libiconv ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/cegui/default.nix:14]:
```suggestion
                ++ stdenv.lib.optionals stdenv.isDarwin [ libiconv ];
```

File: pkgs/development/libraries/ogre/1.10.x.nix [hunk 3]
```diff
@@ -6,7 +6,20 @@
 , libXxf86vm, libICE
 , libXrender
 , withNvidiaCg ? false, nvidia_cg_toolkit
-, withSamples ? false }:
+, withSamples ? false
+, writeScriptBin
+, rsync
+, libobjc
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/ogre/1.10.x.nix:12]: This shouldn't be needed with Foundation.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Fails to build on linux:

```
-- Configuring done
CMake Error at Components/Overlay/CMakeLists.txt:42 (add_library):
  Cannot find source file:

    /build/source/build/imgui-1.77/imgui.cpp

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
  .hpp .hxx .in .txx


CMake Error at Components/Overlay/CMakeLists.txt:42 (add_library):
  No SOURCES given to target: OgreOverlay


CMake Generate step failed.  Build files cannot be regenerated correctly.
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 99727` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>ogre</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>cegui</li>
    <li>ogre1_10</li>
  </ul>
</details>

```
-- Configuring done
CMake Error at Components/Overlay/CMakeLists.txt:42 (add_library):
  Cannot find source file:

    /tmp/nix-build-ogre-1.12.9.drv-0/source/build/imgui-1.77/imgui.cpp

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
  .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .ispc


CMake Error at Components/Overlay/CMakeLists.txt:42 (add_library):
  No SOURCES given to target: OgreOverlay


CMake Generate step failed.  Build files cannot be regenerated correctly.
```

File: pkgs/development/libraries/ogre/1.10.x.nix (hunk 3 above)
Comment (siraben) [reply pkgs/development/libraries/ogre/1.10.x.nix:12]: This is needed because the build fails when it could not find `libobjc`.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 99727` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>cegui</li>
    <li>ogre</li>
    <li>ogre1_10</li>
  </ul>
</details>

**Pull request closed without merging**